tests/test_2024_03_18.py

```diff
@@ -0,0 +1,286 @@
+CSM = "0xdA7dE2ECdDfccC6c3AF10108Db212ACBBf9EA83F"
+VEBO = "0x0De4Ea0184c2ad0BacA7183356Aea5B8d5Bf5c6e"
+AGENT = "0x3e40D73EB977Dc6a537aF587D48316feE66E9C8c"
+VOTING = "0x2e59A20f205bB85a89C53f1936454680651E618e"
+VALIDATORS_EXIT_BUS_ORACLE = "0x0De4Ea0184c2ad0BacA7183356Aea5B8d5Bf5c6e"
+ORACLE_DAEMON_CONFIG = "0xbf05A929c3D7885a6aeAd833a992dA6E5ac23b09"
+WITHDRAWAL_QUEUE = "0x889edC2eDab5f40e902b864aD4d7AdE8E412F9B1"
+CS_ACCOUNTING = "0x4d72BFF1BeaC69925F8Bd12526a39BAAb069e5Da"
+CS_FEE_ORACLE = "0x4D4074628678Bd302921c20573EEa1ed38DdF7FB"
+EASY_TRACK = "0xF0211b7660680B49De1A7E9f25C65660F0a13Fea"
+FINANCE = "0xB9E5CBB9CA5b0d659238807E84D0176930753d86"
+
+DAI_TOKEN = "0x6B175474E89094C44Da98b954EedeAC495271d0F"
+USDT_TOKEN = "0xdAC17F958D2ee523a2206206994597C13D831ec7"
+USDC_TOKEN = "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48"
+
+# Roles
+PAUSE_ROLE = "0x139c2898040ef16910dc9f44dc697df79363da767d8bc92f2e310312b816e46d"
+
+# Vote duration
+OLD_VOTE_DURATION = 259200
+OLD_OBJECTION_PHASE_DURATION = 86400
+
+# Oracle daemon config
+FINALIZATION_MAX_NEGATIVE_REBASE_EPOCH_SHIFT_OLD_VALUE = 1350
+
+
+def _check_role(contract: Contract, role: str, holder: str):
+    role_bytes = web3.keccak(text=role).hex()
+    assert contract.getRoleMemberCount(role_bytes) == 1, f"Role {role} on {contract} should have exactly one holder"
+    assert contract.getRoleMember(role_bytes, 0).lower() == holder.lower(), f"Role {role} holder on {contract} should be {holder}"
+
+
+def _check_no_role(contract: Contract, role: str, holder: str):
+    role_bytes = web3.keccak(text=role).hex()
+    assert contract.getRoleMemberCount(role_bytes) == 1, f"Role {role} on {contract} should have exactly one holder"
+    assert (
+        not contract.getRoleMember(role_bytes, 0).lower() == holder.lower()
+    ), f"Role {role} holder on {contract} should be {holder}"
+
+
+def test_vote(helpers, accounts, vote_ids_from_env, bypass_events_decoding, stranger):
+    acl = interface.ACL(ACL)
+    csm = interface.CSModule(CSM)
+    cs_fee_oracle = interface.CSFeeOracle(CS_FEE_ORACLE)
+    cs_accounting = interface.CSAccounting(CS_ACCOUNTING)
+    vebo = interface.ValidatorsExitBusOracle(VEBO)
+    finance = interface.Finance(FINANCE)
+    agent = interface.Agent(AGENT)
+    voting = interface.Voting(VOTING)
+    easy_track = interface.EasyTrack(EASY_TRACK)
+    withdrawal_queue = interface.WithdrawalQueue(WITHDRAWAL_QUEUE)
+    oracle_daemon_config = interface.OracleDaemonConfig(ORACLE_DAEMON_CONFIG)
+
+    # On-chain voting duration state before voting
+    assert voting.voteTime() == OLD_VOTE_DURATION
+    assert voting.objectionPhaseTime() == OLD_OBJECTION_PHASE_DURATION
+    assert not acl.hasPermission(
+        voting.address,
+        voting.address,
+        voting.UNSAFELY_MODIFY_VOTE_TIME_ROLE()
+    )
+
+    # Check Oracle Config state before voting
+    new_value_uint = convert.to_uint(oracle_daemon_config.get("FINALIZATION_MAX_NEGATIVE_REBASE_EPOCH_SHIFT"))
```

Comment:
please rename to something meaningful, and it is not `new_value` anyway, rather an `old` one